src/sidebar/components/menu-keyboard-navigation.js

```diff
@@ -0,0 +1,118 @@
+import { createElement } from 'preact';
+import { useEffect, useRef } from 'preact/hooks';
+import propTypes from 'prop-types';
+
+import { normalizeKeyName } from '../../shared/browser-compatibility-utils';
+
+function isElementVisible(element) {
+  return element.offsetParent !== null;
+}
+
+/**
+ * Helper component used by Menu and MenuItem to facilitate keyboard navigation of a
+ * list of <MenuItem> components. This component should not be used directly.
+ *
+ * Note that `ArrowRight` shall be handled by the parent <MenuItem> directly and
+ * all other focus() related  navigation is handled here.
```

Comment:
```suggestion
 * all other focus() related navigation is handled here.
```

Extra space removed.